mailpoet/lib/Newsletter/Scheduler/PostNotificationScheduler.php

```diff
@@ -114,18 +128,32 @@ public function createPostNotificationSendingTask(NewsletterEntity $newsletter):
     $lastQueue = $newsletter->getLatestQueue();
     $task = $lastQueue !== null ? $lastQueue->getTask() : null;
     $scheduledAt = $task !== null ? $task->getScheduledAt() : null;
-    if ($scheduledAt && $scheduledAt->format('Y-m-d H:i:s') === $nextRunDate) {
+    if ($scheduledAt && $scheduledAt->format('Y-m-d H:i:s') === $nextRunDate->format('Y-m-d H:i:s')) {
       return null;
     }
 
-    $sendingTask = SendingTask::create();
-    $sendingTask->newsletterId = $newsletter->getId();
-    $sendingTask->status = SendingQueueEntity::STATUS_SCHEDULED;
-    $sendingTask->scheduledAt = $nextRunDate;
-    $sendingTask->save();
+    $sendingTask = new ScheduledTaskEntity();
+    $sendingTask->setType(SendingQueue::TASK_TYPE);
+    $sendingTask->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
+    $sendingTask->setScheduledAt($nextRunDate);
+    $sendingTask->setPriority(ScheduledTaskEntity::PRIORITY_MEDIUM);
+    $this->scheduledTasksRepository->persist($sendingTask);
+    $this->scheduledTasksRepository->flush();
+
+    $sendingQueue = new SendingQueueEntity();
+    $sendingQueue->setNewsletter($newsletter);
+    $sendingQueue->setTask($sendingTask);
+    $this->sendingQueuesRepository->persist($sendingQueue);
+    $this->sendingQueuesRepository->flush();
+    $sendingTask->setSendingQueue($sendingQueue);
+    $this->scheduledTasksRepository->flush();
```

Comment:
I guess this flush is redundant because the sending queue is the owner of the relation.